### ISSUE TITLE
freezing in full screen

### ISSUE THREAD

**Issue by twofoureight** [mpv-player/mpv#3269]
### mpv version and platform

Latest git (22291a2) and El Capitan 10.11.5 on Mid 2015 rMBP

### Reproduction steps

Enter full screen.  If you notice a slight screen tear in the beginning you will encounter the issue, if not playback should be normal. 

Note: If the discrete GPU is used playback will be normal, but sometimes you will notice screen tears with some dropped frames and the video speeding up - this is the point I think where the integrated GPU freezes while the discrete GPU drops frames. 

### Expected behavior

Normal playback. 

### Actual behavior

Video freezing (looks like a slideshow) while audio continues. Video only updates when mouse is moved. The OSC also gets frozen, I tried removing the OSC but the issue still occurs. 

### Log files

The issue occurs sporadically so I was unable to get log files. 

### What it looks like

https://gfycat.com/FormalDefensiveGreathornedowl

I didn't use a config file, only default settings where used.

**Comment by pigoz** [comment]
Is this a duplicate of #2392 ?

If so we basically have no idea how to fix

**Comment by twofoureight** [comment]
I don't think so. Playback works fine for the most part, this randomly happens.

**Comment by phantoriahx** [comment]
Same issue here. 
When the menu bar disappears during fullscreen it either freezes or the fps drops significantly. Rarely (but it happened twice to me) it leads to the complete break down of my macbook pro system (Mid 2015, El Capitan) and I had to force restart. 
When the menu bar reappears, i.e. moving the mouse, it goes back to normal again.
This issue did not exist until updating to El Capitan.
This issue happens quite frequently. If it is normal when I started playing videos, it can go freeze again upon using the keyboard to change the position in the timeline (i.e. +5s or -5s, which also involves the reappearing of menu bar)
To put it simply, every time the menu bar appears and disappears, there's always a chance for mpv to go freeze. So my guess would be that it has something to do with the menu bar (perhaps).

Hope it helps.

**Comment by phantoriahx** [comment]
Please do tell me if I can be any help. I am using the latest version by stolendata, and I have tried these two following commands and it did not work:
--fs-black-out-screens
native-fs

**Comment by phantoriahx** [comment]
Further noted:
adding `<vo=opengl:icc-profile-auto>` to config seems to decrease the chance of freezing, as opposed to `<vo=opengl-hq:icc-profile-auto>`. And some videos are more likely to freeze than others.

**Comment by wm4** [comment]
If anyone gets a freeze, can you attach a debugger and backtrace all threads?

Unfortunately I can't give any instructions, because gdb doesn't work on OSX.

**Comment by twofoureight** [comment]
For what it's worth, I fixed this issue by increasing the resolution scale from the default.
